sdks/python/apache_beam/typehints/native_type_compatibility.py

```diff
@@ -144,11 +154,17 @@ def convert_to_beam_type(typ):
           match=_match_issubclass(typing.Tuple),
           arity=-1,
           beam_type=typehints.Tuple),
-      _TypeMapEntry(
-          match=_match_same_type(typing.Union),
-          arity=-1,
-          beam_type=typehints.Union)
   ]
+  if sys.version_info.major >= 3:
+    type_map.append(
+        _TypeMapEntry(
+            match=_match_is_union_py3, arity=-1, beam_type=typehints.Union))
```

Comment:
I think _match_is_union_py3 works just as well on py27, I'd remove this branching. Also, consider adding `_match_is_any` helper.

Reply:
👍 , there are some updates to the same functions, functionality in [BEAM-6985].